packages/medusa/src/services/cart.ts

```diff
@@ -1462,9 +1462,18 @@ class CartService extends TransactionBaseService {
             relations: ["rule", "rule.conditions", "regions"],
           })
 
-        await this.discountService_
-          .withTransaction(transactionManager)
-          .validateDiscountForCartOrThrow(cart, discounts)
+        try {
+          await this.discountService_
+            .withTransaction(transactionManager)
+            .validateDiscountForCartOrThrow(cart, discounts)
+        } catch (error) {
+          if (
+            error?.code === MedusaError.Codes.CUSTOMER_DISCOUNT_LIMIT_REACHED
+          ) {
+            await this.removeDiscount(cart.id, error.code)
```

Comment:
thought: `removeDiscount` takes as the second argument the code we want to remove. You are passing the error code, which will therefore have no effect. Am I missing something?

Reply:
It would be great to have an integration test covering this scenario.